src/vs/platform/contextkey/browser/contextKeyService.ts

```diff
@@ -112,10 +114,23 @@ class ConfigAwareContextValuesContainer extends Context {
 			} else {
 				const changedKeys: string[] = [];
 				for (const configKey of event.affectedKeys) {
-					const contextKey = `config.${configKey}`;
+					let contextKey = `config.${configKey}`;
 					if (this._values.has(contextKey)) {
+						const value = this._values.get(contextKey);
+
 						this._values.delete(contextKey);
 						changedKeys.push(contextKey);
+
+						if (value === configContextObject) {
+							contextKey += '.';
+
+							for (const key of this._values.keys()) {
+								if (key.startsWith(contextKey)) {
```

Comment:
A loop in a loop is fishy, esp since this code has been tweak many time for performance since it always shows up when profiling startup. Did you measure this?

Reply:
OK, I've updated it to avoid a loop within a loop by switching to regex (I ran a bunch of timings and regex was about 50% faster).